File: internal/commands/delete.go
package commands

import (
	"context"
	"fmt"

	"github.com/brettman/ticktickcli/internal/api"
	"github.com/brettman/ticktickcli/internal/config"
	"github.com/brettman/ticktickcli/internal/project"
	"github.com/fatih/color"
	"github.com/manifoldco/promptui"
	"github.com/spf13/cobra"
)

// NewDeleteCmd creates the delete command
func NewDeleteCmd() *cobra.Command {
	var (
		projectID string
		force     bool
	)

	cmd := &cobra.Command{
		Use:   "delete <task-id>",
		Short: "Delete a task",
		Long: `Delete a task from TickTick.

The task ID can be the full ID or a short ID (first 8 characters).
By default, you will be prompted for confirmation unless --force is used.

Examples:
  ticktick delete abc12345
  ticktick delete abc12345 --force
  ticktick delete abc12345 --project xyz789`,
		Args: cobra.ExactArgs(1),
		RunE: func(cmd *cobra.Command, args []string) error {
			taskID := args[0]
			return runDelete(taskID, projectID, force)
		},
	}

	cmd.Flags().StringVar(&projectID, "project", "", "Project ID (overrides .ticktick file)")
	cmd.Flags().BoolVarP(&force, "force", "f", false, "Skip confirmation prompt")

	return cmd
}

// runDelete handles deleting a task
func runDelete(taskID, projectID string, force bool) error {
	// Load config and check authentication
	cfg, err := config.Load()
	if err != nil {
		return fmt.Errorf("failed to load config: %w", err)
	}

	if !cfg.IsAuthenticated() {
		return fmt.Errorf("not authenticated. Run 'ticktick auth login' first")
	}

	// Determine which project to use
	if projectID == "" {
		// Try to find .ticktick file
		ctx, err := project.GetCurrentProjectContext()
		if err == nil {
			projectID = ctx.File.ProjectID
		} else if err == project.ErrNoTickTickFile {
			// Use default project from config
			if cfg.Preferences.DefaultProject != "" {
				projectID = cfg.Preferences.DefaultProject
			} else {
				return fmt.Errorf("no project specified. Run 'ticktick init' or use --project flag")
			}
		} else {
			return fmt.Errorf("failed to load project context: %w", err)
		}
	}

	// Create API client
	client := api.NewClientFromConfig(
		cfg.Auth.ClientID,
		cfg.Auth.ClientSecret,
		cfg.Auth.AccessToken,
		cfg.Auth.RefreshToken,
		cfg.Auth.Expiry,
	)

	ctx := context.Background()

	// Find the task by ID (handle short IDs)
	fullTaskID, taskTitle, err := findTaskByID(ctx, client, projectID, taskID)
	if err != nil {
		return err
	}

	// Confirm deletion unless --force is used
	if !force {
		prompt := promptui.Prompt{
			Label:     fmt.Sprintf("Delete task '%s'", taskTitle),
			IsConfirm: true,
		}

		result, err := prompt.Run()
		if err != nil || (result != "y" && result != "Y") {
			fmt.Println("Deletion cancelled.")
			return nil
		}
	}

	// Delete the task
	if err := client.DeleteTask(ctx, projectID, fullTaskID); err != nil {
		return fmt.Errorf("failed to delete task: %w", err)
	}

	// Display success
	green := color.New(color.FgGreen, color.Bold)
	green.Println("✓ Task deleted!")
	fmt.Printf("\nTask: %s\n", taskTitle)
	fmt.Printf("ID: %s\n", fullTaskID)

	return nil
}

File: src/commands/delete.ts
import { Command } from 'commander';
import chalk from 'chalk';
import inquirer from 'inquirer';
import { ConfigManager } from '../lib/config.js';
import { ProjectManager } from '../lib/project.js';
import { TickTickClient } from '../lib/api-client.js';

export const deleteCommand = new Command('delete')
  .description('Delete a task')
  .argument('<task-id>', 'Task ID (full or short ID)')
  .option('--project <id>', 'Project ID (overrides .ticktick file)')
  .option('-f, --force', 'Skip confirmation prompt')
  .action(async (taskId, options) => {
    try {
      const config = await ConfigManager.load();
      if (!ConfigManager.isAuthenticated(config)) {
        throw new Error('Not authenticated. Run \'ticktick auth login\' first');
      }

      // Determine project
      let projectId = options.project;
      if (!projectId) {
        const ctx = await ProjectManager.getCurrentProjectContext();
        if (ctx) {
          projectId = ctx.file.projectId;
        } else if (config.preferences.defaultProject) {
          projectId = config.preferences.defaultProject;
        } else {
          throw new Error('No project specified. Run \'ticktick init\' or use --project flag');
        }
      }

      const client = new TickTickClient(config.auth.accessToken);

      // Find task by ID
      const task = await client.findTaskById(projectId, taskId);
      if (!task) {
        throw new Error(`Task not found with ID: ${taskId}`);
      }

      // Confirm deletion unless --force
      if (!options.force) {
        const { confirmed } = await inquirer.prompt([
          {
            type: 'confirm',
            name: 'confirmed',
            message: `Delete task '${task.title}'?`,
            default: false,
          },
        ]);

        if (!confirmed) {
          console.log('Deletion cancelled.');
          return;
        }
      }

      // Delete task
      await client.deleteTask(projectId, task.id);

      console.log(chalk.green.bold('✓ Task deleted!'));
      console.log(`\nTask: ${task.title}`);
      console.log(`ID: ${task.id}`);
    } catch (error) {
      console.error(chalk.red('Error:'), (error as Error).message);
      process.exit(1);
    }
  });
